### PR TITLE
Test using concierge

### PR DESCRIPTION
**Pull request by addyess**
### Overview
Replaces the typescript engine in this repo with machinery provided via concierge

### Details
* builds a concierge plan and updates the runner with concierge
* tests lxd across latest/stable, 5.21/stable, and 6.1/stable
* test juju across 2.9/stable, 3.5/stable, 3/stable (dropping 3.1, 3.3, 3.4)

### CODE REVIEW

File: src/bootstrap/main.sh [hunk 1]
```diff
@@ -1,6 +1,112 @@
...
+  lxd:
+    enable: true
+    channel: ${lxd_used_channel}
+EOF
+    fi
+    cat <<EOF >> concierge.yaml
+host:
+  snaps:
+    - charm/${charm_channel}
+    - charmcraft/${charmcraft_channel}
```
Comment (jnsgruk) [inline src/bootstrap/main.sh:94]: Better off setting the `CONCIERGE_CHARMCRAFT_CHANNEL` env var or passing `--charmcraft-channel` think

File: src/bootstrap/main.sh [hunk 2]
```diff
@@ -1,6 +1,112 @@
...
+    - juju/${juju_channel}
```
Comment (jnsgruk) [inline src/bootstrap/main.sh:95]: As above with charmcraft

File: src/bootstrap/main.sh [hunk 3]
```diff
@@ -1,6 +1,112 @@
...
+    - jq/${jq_channel}
```
Comment (jnsgruk) [inline src/bootstrap/main.sh:96]: This seems like unnecessary configurability...

File: src/bootstrap/main.sh [hunk 4]
```diff
@@ -1,6 +1,112 @@
...
+    - juju-bundle/${juju_bundle_channel}
+    - juju-crashdump/${juju_crashdump_channel}
+EOF
+    echo "Concierge plan:"
+    cat concierge.yaml
+    echo "Running concierge..."
+    concierge prepare -c concierge.yaml --trace
```
Comment (jnsgruk) [inline src/bootstrap/main.sh:103]: You can drop the `-c concierge.yaml` as that's the default path it looks for

File: src/bootstrap/main.sh [hunk 5]
```diff
@@ -1,6 +1,113 @@
...
+    - lxd
+    - charm/${charm_channel}
+    - jq/${jq_channel}
+    - juju-bundle/${juju_bundle_channel}
+    - juju-crashdump/${juju_crashdump_channel}
+EOF
+    echo "Concierge plan:"
+    cat concierge.yaml
+    echo "Running concierge..."
+    SHELL=/bin/bash concierge prepare --trace -v
```
Comment (jnsgruk) [inline src/bootstrap/main.sh:104]: I think an explicit `export SHELL=/bin/bash` is needed here, because concierge starts a subshell

**Comment by addyess** [comment]
At this point concierge is stable at running all the tests

**Pull request merged**